lib/auth/auth_with_roles.go

```diff
@@ -833,12 +869,23 @@ func (a *ServerWithRoles) checkAccessToResource(resource types.Resource) error {
 		return a.checkAccessToApp(r.GetApp())
 	case types.DatabaseServer:
 		return a.checkAccessToDatabase(r.GetDatabase())
+	case types.Server:
+		if resource.GetKind() == types.KindKubeService {
+			// KubeServices are present inside the type Server. We need to
+			// filter out the ones that the user doesn't have access to. This
+			// function won't return an AccessDenied because we're not checking
+			// access to the Server itself.
+			return a.filterKubeServices(r)
+		}
+
+		return a.checkAccessToNode(r)
```

Comment:
Since this returns an AccessDenied error, what would the behavior be if the user doesn't have access to any nodes? Will they get an access denied error or an empty list?

Reply:
They will receive and empty. This works the same way for all resources that the `ListResources` call can return.